common/parallel/TreeCrawler.go

```diff
@@ -80,8 +116,25 @@ func (c *crawler) start(ctx context.Context, root Directory) {
 	}
 	go heartbeat()
 
-	c.unstartedDirs = append(c.unstartedDirs, root)
+	if c.isSync && !c.isSource {
+		// Go routine receives dir entries on tqueue, which till now processed by source traverser.
+		go c.receiveOnTqueue()
+
+		// Target traverser follow source traverser, so it needs to wait for directories completed by source traverser.
+		// Target traverse can't come out until source traverser will tell no more entries left. It tells by closing the source and
+		// target communication channel tqueue.
+		c.mayHaveMoreDirs = true
```

Comment:
to be safe against go routine scheduling order, let's move this before the call to receiveOnTqueue above.

Reply:
Yes make sense.